File: src/Ddd/Trainings/Proposal/Domain/ProposalId.cs
using System;
using Ddd.Utils;

namespace Ddd.Trainings.Proposal.Domain
{
    public class ProposalId(string value) : BaseId
    {
    private readonly string _value = value;

    public static ProposalId Create()
        {
            return new ProposalId(Guid.NewGuid().ToString());
        }

    public override bool Equals(object? obj)
    {
      return obj is ProposalId id &&
             _value == id._value;
    }

    public override int GetHashCode()
    {
      return HashCode.Combine(_value);
    }

    public override string GetValue()
    {
      return _value;
    }
  }
} 

File: src/main/java/com/ddd/trainings/proposal/domain/ProposalId.java
package com.ddd.trainings.proposal.domain;

import com.ddd.utils.BaseId;
import java.util.UUID;
import lombok.EqualsAndHashCode;
import lombok.Getter;

@EqualsAndHashCode(of = "value", callSuper = false)
public class ProposalId extends BaseId {

  @Getter
  private final String value;

  public ProposalId(String value) {
    this.value = value;
  }

  public static ProposalId create() {
    return new ProposalId(UUID.randomUUID().toString());
  }
}
